main.cpp

```diff
@@ -1,243 +1,27 @@
-          case (Card::Suit::club)   :    card_string += "C" ;         break;
-          case (Card::Suit::diamond):    card_string += "D" ;         break;
-          case (Card::Suit::heart)  :    card_string += "H" ;         break;
-          default:
-             card_string = "?";         break;
-      }
-
-      // Increment iterator here, that iterates over shuffled deck vector. printCard() calls right after getCardValue(), which needs to occur before iterating to next card in deck that is ready to be drawn
-      m_it++;
-      return card_string;
-  }
-};
-
-class UnitPlayer
-{
-protected:
-  int m_score{};
-  const int c_MAXSCORE{21};
-  std::string m_name{};
-public:
-  UnitPlayer(std::string_view name)
-    : m_name{name}
-  {
-  }
-
-  friend std::ostream& operator<<(std::ostream& out, const UnitPlayer& p)
-  {
-    return p.print(out);
-  }
-
-  virtual std::ostream& print(std::ostream& out) const = 0;
-  virtual void drawCard() = 0;
-
-  std::string_view getName() const{return m_name;}
-  int getScore() const{return m_score;}
-};
-
-class Player : public UnitPlayer
-{
-private:
-  Deck m_deck{};
-  std::vector<std::string> Hand{};
-public:
-  Player(std::string_view name)
-    : UnitPlayer{name}
-  {
-  }
-
-  bool hasBust()  
-  {
-    return (m_score > c_MAXSCORE);
-  }
-  bool wantsHit()  // returns true if player wants a hit
-  {
-    char answer{};
-    std::cout << "Do you want to hit (H / h) or stay (S / s)?";
-    std::cin >> answer;
-    assert((answer == 's' || answer == 'S' || answer == 'h' || answer == 'H') && "Must select h/H/s/S\n");
-    return (answer == 'H' || answer == 'h');
-  }
-
-  std::ostream& print(std::ostream& out) const override
-  {
-    out << "Player" << m_name << " has a score of << " << m_score << "\nHas cards: \n";
-    int it{};
-    for(const auto& cardStrings : Hand)
-    {
-      out << cardStrings.at(it) << " ";
-      it++;
-    }
-    out << '\n';
-    return out;
-  }
-
-  void updateScore(int value){m_score++;}
-
-  void drawCard() override
-  {
-    updateScore(m_deck.getCardValue());
-    Hand.push_back(m_deck.printCard());
-  }
-
-};
-
-class Dealer : public UnitPlayer
-{
-private:
-  Deck m_deck{};
-  std::vector<std::string> Hand{};
-public:
-  Dealer(std::string_view name)
-    : UnitPlayer{name}
-  {
-  }
-
-  std::ostream& print(std::ostream& out) const override
-  {
-    out << "Dealer " << m_name << " has a score of << " << m_score << "\nHas cards: \n";
-    int it{};
-    for(const auto& cardStrings : Hand)
-    {
-      out << cardStrings.at(it) << " ";
-      it++;
-    }
-    out << '\n';
-    return out;
-  }
-
-  void updateScore(int value){m_score++;}
-
-  void drawCard() override
-  {
-    updateScore(m_deck.getCardValue());
-    Hand.push_back(m_deck.printCard());
-  }
-
-};
-
-
+#include <memory> // smart pointers
 
 int main()
 {
-  std::vector <UnitPlayer*> v
+  auto dealer{std::make_unique<Dealer>("Dan")};
```

Comment:
So, I think from an academic standpoint using curly braces to initialize things is preferred. But, in professional code I think everyone prefers `auto dealer = std::make_unique...`. I find the curly brace initialization ugly and only use it in the niche circumstances it has a meaningful benefit. But I remember reading Bjarne's book and he used it a lot too so that why I surmised it may be popular in an academic sense

Reply:
Interesting. I'd say if that is the consensus then I ought to change it